db/job.go

```diff
@@ -913,3 +657,109 @@ func scanJobs(conn Conn, lockFactory lock.LockFactory, rows *sql.Rows) (Jobs, er
 
 	return jobs, nil
 }
+
+func (j *job) SyncResourceSpaceCombinations(combinations []map[string]string) ([]JobCombination, error) {
+	tx, err := j.conn.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	defer Rollback(tx)
+
+	jobCombinations := []JobCombination{}
+
+	for _, c := range combinations {
+		var jobCombinationID int
+
+		marshaled, err := json.Marshal(c)
+		if err != nil {
+			return nil, err
+		}
+
+		needsMigration, err := findInvalidJobCombination(tx, j.ID())
+		if err != nil {
+			return nil, err
+		}
+
+		if needsMigration {
```

Comment:
It would be nice if we could isolate this to the migration, but that's blocked on concourse/concourse#1973.